### PR TITLE
Initial typing in client.py

### PR DESCRIPTION
**Pull request by OskarBrzeski**
Not fully certain about some of the type hints, especially regarding `Client.request()`.

### CODE REVIEW

**Comment by lavigne958** [comment]
Hi you can ise the following command to format the entire code and pass the linter: `tox -e format`

**Comment by OskarBrzeski** [comment]
The issue seems to be with the imports in exceptions.py and cell.py. Should I commit those changes to this PR?

Also, tox sees this line `passenv = GS_RECORD_MODE GS_CREDS_FILENAME APPDATA` in tox.ini as an error because it wants commas instead of spaces. Should I change this too?

**Comment by lavigne958** [comment]
I see, I did not finish my fishes on `gspread/utils.py` but I have the fix for those errors on the files you did not change. I will push the fixes on those files so you can rebase your pr (or merge the branch `feature/release_6_0_0` in your branch or cherry-pick the commits) then it should pass the CI test.

I'll let you know when you can rebase. sorry for the inconvenience

**Comment by lavigne958** [comment]
:heavy_check_mark:  I pushed the necessary changes to make the CI green again on the branch `feature/release_6_0_0`. you can rebase your branch onto it.

**Review by lavigne958 — COMMENTED** [review]
thank you for this contribution, here are a few comments, mostly about JSON to be typed as Any because it's a nightmare to maintain.

File: gspread/client.py [hunk 1]
```diff
@@ -8,9 +8,11 @@
 """
 
 from http import HTTPStatus
-from typing import Type
+from typing import IO, Any, Dict, List, Mapping, Optional, Tuple, Type, Union
 
+from google.auth.credentials import Credentials
 from google.auth.transport.requests import AuthorizedSession
```
Comment (lavigne958) [inline gspread/client.py:14]: last thing we can do here is using the special mypy comment to ignore these missing import: `# type: ignore` as a comment at the end of the line.

don't forget to format your code after that :wink:

**Review by lavigne958 — COMMENTED** [review]
last changes

File: gspread/client.py (hunk 1 above)
Comment (OskarBrzeski) [reply gspread/client.py:14]: Should we do this for the requests package too?

Comment (lavigne958) [reply gspread/client.py:14]: I don't see any issues with the `requests` package. Mypy apparently found the typing stubs

**Review by lavigne958 — APPROVED** [review]
Awesome. Looks good to me. I'll run mypy on it when I get z Chance

**Pull request merged**